File: 219-contains-duplicate-II/Solution.java
import java.util.HashMap;
import java.util.Map;

class Solution {

    public static void main(String[] args) {


        int[] arrayOne =  {1,2,3,1};
        int[] arrayTwo =  {1,0,1,1};
        int[] arrayThree =  {1,2,3,1,2,3};

        System.out.println(containsNearbyDuplicate(arrayOne, 3));
        System.out.println(containsNearbyDuplicate(arrayTwo, 1));
        System.out.println(containsNearbyDuplicate(arrayThree, 2));
    }

    public static boolean containsNearbyDuplicate(int[] nums, int k) {
        
        Map<Integer, Integer> mp = new HashMap<>();

        int n = nums.length;

        for(int i = 0; i < n; i++) {

            if(mp.get(nums[i]) != null) {
                if(i - mp.get(nums[i]) <= k) {
                     return true;
                }
            }

            mp.put(nums[i], i);
        }


        return false;
    }
}

File: 219-contains-duplicate-II/Solution.py
from typing import List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        n = len(nums)
        map = {}

        for idx in range(0,n):

            if nums[idx] in map:
                if idx - map[nums[idx]] <= k:
                    return True
                
            map[nums[idx]] = idx
        
        return False

arr = [1,2,3,1]

obj = Solution()

print(obj.containsNearbyDuplicate(arr, 3))

